Up2dateService/Up2dateService/SetupManager/ChocoNugetInfo.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Text;
+using System.Xml;
+
+namespace Up2dateService.SetupManager
+{
+    public class ChocoNugetInfo
+    {
+        public ChocoNugetInfo(string id, string title, string version, string publisher)
+        {
+            if (string.IsNullOrWhiteSpace(id))
+                throw new ArgumentException($@"'{nameof(id)}' cannot be null or whitespace.", nameof(id));
+
+            Id = id;
+            Title = title;
+            Version = version;
+            Publisher = publisher;
+        }
+
+        public string Id { get; }
+        public string Title { get; }
+        public string Version { get; }
+        public string Publisher { get; }
+
+        public static ChocoNugetInfo GetInfo(string fullFilePath)
+        {
+            using (var zipFile = ZipFile.OpenRead(fullFilePath))
+            {
+                var nuspec = zipFile.Entries.First(zipArchiveEntry => zipArchiveEntry.Name.Contains(".nuspec"));
+                using (var nuspecStream = nuspec.Open())
+                {
+                    using (var sr = new StreamReader(nuspecStream, Encoding.UTF8))
+                    {
+                        var xmlData = sr.ReadToEnd();
+                        var doc = new XmlDocument();
+                        doc.LoadXml(xmlData);
+                        var id = doc.GetElementsByTagName("id")[0] != null
```

Comment:
Still throwing exception if tag is absent.
To avoid this replace
doc.GetElementsByTagName("...")[0] != null
with
doc.GetElementsByTagName("...").Count > 0

Reply:
fixed